examples/encodeRadioDetails/encodeRadioDetails.ino

```diff
@@ -0,0 +1,265 @@
+} // setup
+
+/*
+      encoded_details packing order
+
+      1 csn_pin
+      2 ce_pin
+      3 spi_speed
+      4 getChannel()
+      5 getDataRate()
+      6 getPALevel()
+      7 ((read_register(RF_SETUP) & 1) * 1)
+      8 getCRCLength()
+      9 ((read_register(SETUP_AW) & 3) + 2)
+      10 getPayloadSize()
+      11 (read_register(SETUP_RETR) >> ARD)
+      12 (read_register(SETUP_RETR) & 0x0F)
+      13 (read_register(OBSERVE_TX) >> 4)
+      14 (read_register(OBSERVE_TX) & 0x0F)
+      15 static_cast<bool>(read_register(FEATURE) & _BV(EN_DYN_ACK))
+      16 static_cast<bool>(read_register(FEATURE) & _BV(EN_ACK_PAY))
+      17 ((read_register(DYNPD) && (read_register(FEATURE) &_BV(EN_DPL))) * 1)
+      18 (static_cast<bool>(temp_8_bit & _BV(ENAA_P5)) + 48)
+      19 (static_cast<bool>(temp_8_bit & _BV(ENAA_P4)) + 48)
+      20 (static_cast<bool>(temp_8_bit & _BV(ENAA_P3)) + 48)
+      21 (static_cast<bool>(temp_8_bit & _BV(ENAA_P2)) + 48)
+      22 (static_cast<bool>(temp_8_bit & _BV(ENAA_P1)) + 48)
+      23 (static_cast<bool>(temp_8_bit & _BV(ENAA_P0)) + 48)
+      24 (read_register(NRF_CONFIG) & _BV(PRIM_RX))
+      // TX_ADDR
+      25 forty_bit_register_array[0]
+      26 forty_bit_register_array[1]
+      27 forty_bit_register_array[2]
+      28 forty_bit_register_array[3]
+      29 forty_bit_register_array[4]
+      // pipe 0 address
+      30 forty_bit_register_array[0]
+      31 forty_bit_register_array[1]
+      32 forty_bit_register_array[2]
+      33 forty_bit_register_array[3]
+      34 forty_bit_register_array[4]
+      // pipe 1 address
+      35 forty_bit_register_array[0]
+      36 forty_bit_register_array[1]
+      37 forty_bit_register_array[2]
+      38 forty_bit_register_array[3]
+      39 forty_bit_register_array[4]
+      // pipes 2-5 addresses
+      40 eight_bit_register_array[0]
+      41 eight_bit_register_array[0]
+      42 eight_bit_register_array[0]
+      43 eight_bit_register_array[0]
+      44 read_register(EN_RXADDR)
+*/
+
+/*
+  https://www.rapidtables.com/convert/number/binary-to-decimal.html
+  Enter in your binary number and this will convert it to decimal (base10) for you
+  a uint16_t has 16 bits, a uint8_t has 8 bits, a boolean value is packed into one bit
+  in the bit array.  Trailing zeros are padding
+*/
+
+void loop() {
+  if (role) {
+    uint8_t split_payload_2d_array[2][21] = {{0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 99}, {0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 101}};  // split the payload to get below 32 bytes payload limit, mark the end for reassembly
+    for (int i = 0; i < 20; i++)
+    {
+      split_payload_2d_array[0][i] = payload[i]; // first 20 elements
+      split_payload_2d_array[1][i] = payload[i]; // second 20 elements
+    }
+    for (int i = 0; i < 2; i++) // iterate twice to transmit, because we split our payload into two pieces
```

Comment:
Its more performant to use `++i` in for loops because of the difference in how pre/post `++` operators are overloaded.

Reply:
> @dstroy0 put on the brakes!
> 
> The more I think about it, I'm having trouble getting behind these additions.
> 
>     1. Transmitting radio details to another nRF24 would require that the configuration of both radios should match. So, it doesn't make any practical sense to transmit debug info from 1 device to another.
>        
>        * I know you were thinking in terms of networking, but the configuration for network nodes is even more restricted to a certain criteria.
> 
>     2. These changes require a lot more code with very little in the way of optimization.
>        
>        * needs another buffer allocated to store debug info
>        * packing the debug data into another buffer then unpacking it into a char buffer seems redundant
>        * the CPU cycles spent on doing bit manipulation may actually outweigh the memory benefit of manipulating bits
>        * the added namespace of functions is likely increasing the compile size much more than the amount of memory being spared
> 
>     3. I think `sprintf()` could be used better because it returns the number bytes written to the specified buffer. This returned data could be used for subsequent calls to `sprintf()` because the number of bytes written does not include the terminating null byte that is automatically appended.
>        ```c++
>        uint16_t bytes_used = 0;
>        char* buf = new char[870];
>        bytes_used += sprintf(buf, "some %s data", "string");
>        sprintf(buf + bytes_used, "some number %d", 42);
>        ```
>        
>        
>            
>              
>            
>        
>              
>            
>        
>            
>          
>        
>        * I think the original `printPrettyDetails()` could be made to reuse `sprintfPrettyDetails()`, but this idea might not be any more efficient than using `printf()` directly.
> 
> 
> I mentioned this in your previous PR, but I'll say it again because we're still discussing the concepts of these changes. Generally, we discuss new additions/changes in a new [issue](https://github.com/nRF24/RF24/issues) before opening a PR and spending time revising code. I feel like I'm aimlessly reviewing these changes because I'm not sure what this PR is designed to satisfy.
> 
>     * sending me a personal email is not a good alternative because I'm 1 person in a community working to develop/maintain these libraries. Directing the conversation to only me excludes the rest of the people with vested interest in this library.

The output method encodeRadioDetails() uses much less ram and flash than other output methods available in the library when used on its own, that alone makes it worth pursuing.  I also don't understand the point of having two sprintf output methods either, it's there to make debugging encodeRadioDetails() easier and that's it right now, it's pointless.